utils/python/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -0,0 +1,357 @@
+        self._common_setup()
+        self._case_one_setup()
+        # Flush the case so that, if errors occur later, then at least case 1 is
+        # in a correct, post-setup state. This is important because the mere
+        # existence of a case 2 directory signals that setup is done. So if the
+        # build fails and the user rebuilds, setup won't be redone - so it's
+        # important to ensure that the results of setup are flushed to disk.
+        #
+        # Note that case 1 will be in its post-setup state even if case 2 setup
+        # fails. Putting the case1 flush after case 2 setup doesn't seem to help
+        # with that (presumably some flush is called automatically), and anyway
+        # wouldn't help with things like appending to user_nl files (which don't
+        # rely on flush). So we just have to live with that possibility (but
+        # note that we print a warning to the log file if that happens, in the
+        # caller of this method).
+        self._case.flush()
+
+        # Set up case 2
+        self._activate_case2()
+        self._common_setup()
+        self._case_two_setup()
+        # Flush the case so that, if errors occur later, then at least case2 is
+        # in a correct, post-setup state
+        self._case.flush()
+
+        # Go back to case 1 to ensure that's where we are for any following code
+        self._activate_case1()
+
+    def _force_case2_settings(self):
+        """
+        Sets some settings in case2 that are normally set automatically.
+
+        This is needed because we aren't running case2 via the normal mechanism
+        (i.e., via the submit script).
+        """
+
+        # RUN_WITH_SUBMIT is normally set when you run the case's submit script.
+        # Trick the scripts into thinking that we are running via the submit
+        # script, like we're supposed to
+        self._case2.set_value("RUN_WITH_SUBMIT",True)
+
+    def _link_to_case2_output(self):
+        """
+        Looks for all files in rundir2 matching the pattern casename2*.nc.run2suffix
+
+        For each file found, makes a link in rundir1 pointing to this file; the
+        link is renamed so that the original occurrence of casename2 is replaced
+        with casename1.
+
+        For example:
+
+        /glade/scratch/sacks/somecase/run/somecase.clm2.h0.nc.run2 ->
+        /glade/scratch/sacks/somecase.run2/run/somecase.run2.clm2.h0.nc.run2
+
+        If the destination link already exists and points to the correct
+        location, it is maintained as is. However, an exception will be raised
+        if the destination link is not exactly as it should be: we avoid
+        overwriting some existing file or link.
+        """
+
+        casename1 = self._case1.get_value("CASE")
+        casename2 = self._case2.get_value("CASE")
+        rundir1 = self._case1.get_value("RUNDIR")
+        rundir2 = self._case2.get_value("RUNDIR")
+        run2suffix = self._run_two_suffix
```

Comment:
I won't be able to have an informed opinion until I try to use this framework to set up a test. That said, I'm a little concerned at the size and complexity (14 methods!) of this class when the goal was to reduce code size. It does appear that your lii.py test was made very easy under this framework, so that's good. If you can get similar benefits on several of the other tests then I guess I'd be comfortable with this.